rfcs/0185-redistribute-redistributable.md

```diff
@@ -0,0 +1,225 @@
+---
+feature: redistribute-redistributable
+start-date: 2024-12-15
+author: Ekleog
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: https://github.com/NixOS/nixpkgs/issues/83884
+---
+
+# Summary
+[summary]: #summary
+
+Make Hydra build and provide all redistributable software, while making sure installation methods stay as fully free as today.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, Hydra builds only free software and unfree redistributable firmware.
+This means that unfree redistributable software needs to be rebuilt by all the users.
+For example, using MongoDB on a Raspberry Pi 4 (aarch64, which otherwise has access to hydra's cache) takes literally days and huge amounts of swap.
+
+Hydra could provide builds for unfree redistributable software, at minimal added costs.
+This would make life much better for users of such software.
+Especially when the software is still source-available even without being free software, like MongoDB.
+
+# Detailed design
+[design]: #detailed-design
+
+We will add a `runnableOnHydra` field on all licenses, that will be initially set to its `free` field, and set to `true` only for well-known licenses.
+
+Hydra will build all packages with licenses for which `redistributable && runnableOnHydra`.
+It will still fail evaluation if the ISO image build or the Amazon AMIs were to contain any unfree software.
+
+This will be done by evaluating Nixpkgs twice in `release.nix`.
+Once with `allowUnfree = false` like today, plus once with `allowlistedLicenses = builtins.filter (l: l.redistributable && l.runnableOnHydra) lib.licenses`.
+Then, most of the jobs will be taken from the allowlisted nixpkgs, while only the builds destined for installation will be taken from the no-unfree nixpkgs.
+
+The list of jobs destined for installation, that cannot contain unfree software is:
+- `amazonImage`
+- `amazonImageAutomaticSize`
+- `amazonImageZfs`
+- `iso_gnome`
+- `iso_minimal`
+- `iso_minimal_new_kernel`
+- `iso_minimal_new_kernel_no_zfs`
+- `iso_plasma5`
+- `iso_plasma6`
```

Comment:
nit: The graphical ISOS were unified, and Plasma 5 is gone. But I don’t know if it’s worth listing them explicitly anyway.